Mathlib/MeasureTheory/Function/LpSeminorm/ChebyshevMarkov.lean

```diff
@@ -15,58 +15,102 @@ open scoped NNReal ENNReal
+    ε * μ { x | ε ≤ (‖f x‖₊ : ℝ≥0∞) ^ p.toReal } ≤ eLpNorm f p μ ^ p.toReal :=
+  mul_meas_ge_le_pow_eLpNorm_enorm μ hp_ne_zero hp_ne_top hf ε
 
 /-- A version of Chebyshev-Markov's inequality using Lp-norms. -/
-theorem mul_meas_ge_le_pow_eLpNorm' (hp_ne_zero : p ≠ 0) (hp_ne_top : p ≠ ∞)
-    (hf : AEStronglyMeasurable f μ) (ε : ℝ≥0∞) :
-    ε ^ p.toReal * μ { x | ε ≤ ‖f x‖₊ } ≤ eLpNorm f p μ ^ p.toReal := by
-  convert mul_meas_ge_le_pow_eLpNorm μ hp_ne_zero hp_ne_top hf (ε ^ p.toReal) using 4
+theorem mul_meas_ge_le_pow_eLpNorm'_enorm (hp_ne_zero : p ≠ 0) (hp_ne_top : p ≠ ∞)
+    {f : α → ε'} (hf : AEStronglyMeasurable f μ) (ε : ℝ≥0∞) :
+    ε ^ p.toReal * μ { x | ε ≤ ‖f x‖ₑ } ≤ eLpNorm f p μ ^ p.toReal := by
+  convert mul_meas_ge_le_pow_eLpNorm_enorm μ hp_ne_zero hp_ne_top hf (ε ^ p.toReal) using 4
   ext x
   rw [ENNReal.rpow_le_rpow_iff (ENNReal.toReal_pos hp_ne_zero hp_ne_top)]
 
-theorem meas_ge_le_mul_pow_eLpNorm (hp_ne_zero : p ≠ 0) (hp_ne_top : p ≠ ∞)
-    (hf : AEStronglyMeasurable f μ) {ε : ℝ≥0∞} (hε : ε ≠ 0) :
-    μ { x | ε ≤ ‖f x‖₊ } ≤ ε⁻¹ ^ p.toReal * eLpNorm f p μ ^ p.toReal := by
-  by_cases h : ε = ∞
-  · simp [h]
-  have hεpow : ε ^ p.toReal ≠ 0 := (ENNReal.rpow_pos (pos_iff_ne_zero.2 hε) h).ne.symm
-  have hεpow' : ε ^ p.toReal ≠ ∞ := ENNReal.rpow_ne_top_of_nonneg ENNReal.toReal_nonneg h
+/-- A version of Chebyshev-Markov's inequality using Lp-norms. -/
+theorem mul_meas_ge_le_pow_eLpNorm' (hp_ne_zero : p ≠ 0) (hp_ne_top : p ≠ ∞)
+    {f : α → E} (hf : AEStronglyMeasurable f μ) (ε : ℝ≥0∞) :
+    ε ^ p.toReal * μ { x | ε ≤ ‖f x‖₊ } ≤ eLpNorm f p μ ^ p.toReal :=
+  mul_meas_ge_le_pow_eLpNorm'_enorm μ hp_ne_zero hp_ne_top hf ε
+
+theorem meas_ge_le_mul_pow_eLpNorm_enorm (hp_ne_zero : p ≠ 0) (hp_ne_top : p ≠ ∞)
+    {f : α → ε'} (hf : AEStronglyMeasurable f μ) {ε : ℝ≥0∞} (hε : ε ≠ 0) (hε_ne_top : ε ≠ ∞) :
+    μ { x | ε ≤ ‖f x‖ₑ } ≤ ε⁻¹ ^ p.toReal * eLpNorm f p μ ^ p.toReal := by
+  have hεpow : ε ^ p.toReal ≠ 0 := (ENNReal.rpow_pos (pos_iff_ne_zero.2 hε) hε_ne_top).ne.symm
+  have hεpow' : ε ^ p.toReal ≠ ∞ := ENNReal.rpow_ne_top_of_nonneg ENNReal.toReal_nonneg hε_ne_top
   rw [ENNReal.inv_rpow, ← ENNReal.mul_le_mul_left hεpow hεpow', ← mul_assoc,
     ENNReal.mul_inv_cancel hεpow hεpow', one_mul]
-  exact mul_meas_ge_le_pow_eLpNorm' μ hp_ne_zero hp_ne_top hf ε
+  exact mul_meas_ge_le_pow_eLpNorm'_enorm μ hp_ne_zero hp_ne_top hf ε
 
-theorem MemLp.meas_ge_lt_top' {μ : Measure α} (hℒp : MemLp f p μ) (hp_ne_zero : p ≠ 0)
-    (hp_ne_top : p ≠ ∞) {ε : ℝ≥0∞} (hε : ε ≠ 0) :
-    μ { x | ε ≤ ‖f x‖₊ } < ∞ := by
-  apply (meas_ge_le_mul_pow_eLpNorm μ hp_ne_zero hp_ne_top hℒp.aestronglyMeasurable hε).trans_lt
-    (ENNReal.mul_lt_top ?_ ?_)
+-- More general version, allowing ε = ∞ at the expense of an additional condition.
+theorem meas_ge_le_mul_pow_eLpNorm_enorm' (hp_ne_zero : p ≠ 0) (hp_ne_top : p ≠ ∞)
+    {f : α → ε'} (hf : AEStronglyMeasurable f μ)
+    {ε : ℝ≥0∞} (hε : ε ≠ 0) (hmeas_top : μ {x | ‖f x‖ₑ = ⊤} = 0) :
```

Comment:
(unfortunate clash of variable names now that we're using \eps for types...)

Reply:
Common generalization with unprimed version: assume `ε = ∞ -> μ {x | ‖f x‖ₑ = ⊤} = 0`.